examples/medplum-photon-integration/src/bots/sync-patient-from-photon.ts

```diff
@@ -0,0 +1,389 @@
+  PhotonPatient,
+  PhotonPatientAllergy,
+  PhotonPrescription,
+  PhotonProvider,
+} from '../photon-types';
+import { NEUTRON_HEALTH, NEUTRON_HEALTH_PATIENTS } from './constants';
+import { getMedicationElement, handlePhotonAuth, photonGraphqlFetch } from './utils';
+
+export async function handler(medplum: MedplumClient, event: BotEvent): Promise<void> {
+  const photonClientId = event.secrets['PHOTON_CLIENT_ID']?.valueString;
+  const photonClientSecret = event.secrets['PHOTON_CLIENT_SECRET']?.valueString;
+  const photonAuthToken = await handlePhotonAuth(photonClientId, photonClientSecret);
+
+  const query = `
+    query patients(
+      $filter: PatientFilter,
+      $after: ID,
+      $first: Int
+    ) {
+      patients (
+        filter: $filter,
+        after: $after,
+        first: $first
+      ) {
+        id
+        externalId
+        name {
+          title
+          first
+          last
+        }
+        dateOfBirth
+        sex
+        gender
+        email
+        phone
+        allergies {
+          allergen {
+            id
+            name
+            rxcui
+          }
+          comment
+          onset
+        }
+        prescriptions {
+          id
+          externalId
+          prescriber {
+            id
+            externalId
+            name {
+              title
+              first
+              last
+            }
+            email
+            phone
+            address {
+              street1
+              street2
+              postalCode
+              country
+              state
+              city
+            }
+            organizations {
+              id
+              name
+            }
+          }
+          state
+          treatment {
+            id
+            codes {
+              rxcui
+            }
+          }
+          dispenseAsWritten
+          dispenseQuantity
+          refillsAllowed
+          refillsRemaining
+          fillsAllowed
+          fillsRemaining
+          daysSupply
+          instructions
+          notes
+          effectiveDate
+          expirationDate
+          writtenAt
+          fills {
+            id
+            treatment {
+              id
+              codes {
+                rxcui
+              }
+            }
+            state
+            requestedAt
+            filledAt
+          }
+        }
+        address {
+          street1
+          street2
+          postalCode
+          country
+          state
+          city
+        }
+      }
+    }
+  `;
+
+  const body = JSON.stringify({ query });
+
+  const result = await photonGraphqlFetch(body, photonAuthToken);
+  const photonPatients = result.data.patients as PhotonPatient[];
+  const batch: Bundle = {
+    resourceType: 'Bundle',
+    type: 'batch',
+    entry: [],
+  };
+
+  for (const photonPatient of photonPatients) {
+    // Make sure the patient does not already exist
+    if (!(await checkForExistingPatient(photonPatient, medplum))) {
```

Comment:
I think we should still sync the patient's allergies and medications even if the patient already exists. We just don't want to update the `Patient` resource